akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -461,6 +461,12 @@ private[http] object HttpServerBluePrint {
       def emitErrorResponse(response: HttpResponse): Unit =
         emit(responseCtxOut, ResponseRenderingContext(response, closeRequested = true), () ⇒ completeStage())
 
+      def maybePullRequestParsingIn(): Unit =
```

Comment:
Fixing whats broken makes sense—increasing complexity needlessly seems to be at odds with BSR.
@jrudolph If `!hasBeenPulled(requestParsingIn) && isAvailable(requestPrepOut)` does the same job as the var, then I think the var loses. :)